src/promptflow-evals/promptflow/evals/evaluate/_batch_run_client/batch_run_context.py

```diff
@@ -0,0 +1,28 @@
+import os
+
+from promptflow._utils.user_agent_utils import ClientUserAgentUtil
+from promptflow.tracing._integrations._openai_injector import inject_openai_api, recover_openai_api
+
+from ..._user_agent import USER_AGENT
+from .code_client import CodeClient
+from .proxy_client import ProxyClient
+
+
+class BatchRunContext:
```

Comment:
Does this work cross threads?

Reply:
BatchRunContext is not used inside threads; it's at the scope above multi-threading, so it should work.